src/main/java/com/autotune/common/utils/CommonUtils.java

```diff
@@ -281,8 +281,29 @@ public static Timestamp addDays(Timestamp date, int days) {
         return new Timestamp(cal.getTime().getTime());
     }
 
-    public static double percentile(double percentile, List<Double> items) {
-        Collections.sort(items);
-        return items.get((int) Math.round(percentile / 100.0 * (items.size() - 1)));
+    public static double percentile(double percentile, List<Double> values) {
```

Comment:
Not sure why is this custom implementation of percentile needed? What is wrong with the one in the Math package?

Reply:
Current logic is missing interpolation